torch/distributions/distribution.py

```diff
@@ -221,7 +221,7 @@ def _validate_sample(self, value):
             raise ValueError('The value argument must be within the support')
 
     def __repr__(self):
-        param_names = [k for k, _ in self.arg_constraints.items()]
+        param_names = [k for k, _ in self.arg_constraints.items() if k in self.__dict__]
```

Comment:
Nice catch!!. I have one concern though: in MVNs, if we happen to lazily evaluate *all* the parameters (`covariance_matrix`, `scale_tril` and `precision_matrix`), won't they show up redundantly in the `__repr__`?

Reply:
You are right that if all the lazy parameters happen to have been evaluated, they will show up in `__repr__` even if they are redundant (also true for classes with logits and probs)! The downside then is that the output we get isn't guaranteed to be static (though there shouldn't be a reason to expect it to be static), but the upside is that it can aid debugging by letting us know which attributes have been set without having to go through the instance's `__dict__`. Since we are only reporting the size, I think it's fine to report any redundant parameters for the distribution instance. Preventing this would likely require more complex logic than is worth (at least not until we have other use cases for more elaborate ways of registering params).

Reply:
Thanks for the views @neerajprad . I understand the reason, and yes: shapes really shouldn't be an issue.